src/Symfony/Component/VarDumper/Dump.php

```diff
@@ -0,0 +1,160 @@
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @method self to(callable|resource|string $output)               Sets the output destination of the dump: a line callable, an opened stream or an output path.
+ * @method self as(string $format)                                 Sets the output format of the dump: can be "html", "cli", "text" or "env", which uses the `DUMP_AS` env var or the PHP SAPI.
+ * @method self die()                                              Dies after dumping.
+ * @method self withTitle(string $title)                           Sets the title of the dump.
+ * @method self withTrace(bool $trace)                             Enables/disables dumping the stack trace after dumping.
+ * @method self withMaxItems(int $items)                           Limits the number of items past the first level.
+ * @method self withMaxDepth(int $depth)                           Limits the depth of the dump.
+ * @method self withMaxItemsPerDepth(int $max)                     Limits the number of items per depth level.
+ * @method self withMaxStringLength(int $length)                   Limits the number of characters for dumped strings.
+ * @method self withMaxStringWidth(int $width)                     Limits the number of characters per line for dumped strings.
+ * @method self withRefHandles(bool $handles)                      Enables/disables objects' global identifiers dumping.
+ * @method self withCasters(array $casters, bool $replace = false) Adds or replace casters for resources and objects.
+ * @method self withFilter(int $filter)                            A bit field of Caster::EXCLUDE_* constants.
+ */
+class Dump
+{
+    const HTML = 'html';
+    const CLI = 'cli';
+    const TEXT = 'text';
+
+    private $vars;
+    private $dumpOnDestruct;
+    private $params = array(
+        'to' => null,
+        'as' => array('env'),
+        'die' => false,
+//        'withTitle' => array(''),
+//        'withTrace' => array(false),
+        'withMaxItems' => array(2500),
+        'withMaxDepth' => array(-1),
+        'withMaxItemsPerDepth' => array(-1),
+        'withMaxStringLength' => array(-1),
+        'withMaxStringWidth' => array(-1),
+        'withRefHandles' => array(true),
+        'withCasters' => array(array(), false),
+        'withFilter' => array(0),
+    );
+
+    public function __construct(array $vars, $dumpOnDestruct = true)
+    {
+        $this->vars = $vars;
+        $this->dumpOnDestruct = $dumpOnDestruct;
+    }
+
+    public function __destruct()
+    {
+        if ($this->dumpOnDestruct) {
+            $this->dumpOnDestruct = false;
+            $this->dump();
+        }
+    }
+
+    public function dump()
+    {
+        extract($this->params);
+
+        if ($withCasters[1]) {
+            $cloner = new VarCloner($withCasters[0]);
+        } else {
+            $cloner = new VarCloner();
+            $cloner->addCasters($withCasters[0]);
+        }
+        $cloner->setMaxItems($withMaxItems[0]);
```

Comment:
well, ``die`` and ``to`` are already handled a special way and not stored as array. I suggest you to store these fields as scalars and do a special handling in ``__call`` only for the params needing to store arrays (only ``withCasters`` AFAICT). This would make the code more understandable IMO